pkg/reconciler/broker/resources/eventpolicy.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2024 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	eventingduckv1 "knative.dev/eventing/pkg/apis/duck/v1"
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	eventingv1alpha1 "knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+)
+
+const (
+	BackingChannelEventPolicyLabelPrefix = "eventing.knative.dev/"
+	OIDCBrokerSub                        = "system:serviceaccount:knative-eventing:mt-broker-ingress-oidc"
+	brokerAPIVersion                     = "eventing.knative.dev/v1"
+	version                              = "v1"
+	brokerKind                           = "Broker"
+	brokerGroup                          = "eventing.knative.dev"
+)
+
+func MakeEventPolicyForBackingChannel(b *eventingv1.Broker, backingChannel *eventingduckv1.Channelable) *eventingv1alpha1.EventPolicy {
+	return &eventingv1alpha1.EventPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: backingChannel.Namespace,
+			Name:      b.Name + "-event-policy",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: brokerAPIVersion,
+					Kind:       brokerKind,
+					Name:       b.Name,
+				},
+			},
+			Labels: LabelsForBackingChannelsEventPolicy(b),
+		},
+		Spec: eventingv1alpha1.EventPolicySpec{
+			To: []eventingv1alpha1.EventPolicySpecTo{
+				{
+					Ref: &eventingv1alpha1.EventPolicyToReference{
+						APIVersion: backingChannel.APIVersion,
+						Kind:       backingChannel.Kind,
+						Name:       backingChannel.Name,
+					},
+				},
+			},
+			From: []eventingv1alpha1.EventPolicySpecFrom{
+				{
+					Sub: toStrPtr(OIDCBrokerSub),
+				},
+			},
+		},
+	}
+}
+
+func LabelsForBackingChannelsEventPolicy(broker *eventingv1.Broker) map[string]string {
+	return map[string]string{
+		BackingChannelEventPolicyLabelPrefix + "broker-group":   brokerGroup,
+		BackingChannelEventPolicyLabelPrefix + "broker-version": version,
+		BackingChannelEventPolicyLabelPrefix + "broker-kind":    brokerKind,
+		BackingChannelEventPolicyLabelPrefix + "broker-name":    broker.Name,
```

Comment:
I'm curious if we actually need all of these labels for the EventPolicies, what exactly are you using them all for?

Reply:
@Cali0707 I took reference from channel - https://github.com/knative/eventing/blob/main/pkg/reconciler/channel/resources/eventpolicy.go#L69
Hence, thought of adding the same here. LMK if you think it shouldn't be there, I will remove it !

Reply:
Yeah looking there it seems like those labels are used to correctly select the EventPolicies which were created by the channel. We should also use the labels to make a label selector when checking which EventPolicies are owned by the broker. See here for how it is used in the channel: https://github.com/knative/eventing/blob/bb2e0a30ff68c7f145ef243451c446e1a471b3dc/pkg/reconciler/channel/channel.go#L142-L166

Reply:
gotcha. So just to clarify: currently we are filtering EventPolicies based on the Owner References of broker, instead filter out based on the labels, right ?

Reply:
No, filtering the eventpolicies by ownerreference is good, but when we list the eventpolicies, we should use a label selector with these labels (see line 152 in the code I linked above). So, to summarize what we would do is:
1. List the EventPolicy resources using a label selector to only retrieve relevant ones
2. Filter the EventPolicy resources by owner reference

Reply:
done, PTAL!